### PR TITLE
otelzap: Implement Enabled method

### PR DESCRIPTION
**Pull request by khushijain21**
Part of https://github.com/open-telemetry/opentelemetry-go-contrib/issues/5191

Pre-work https://github.com/open-telemetry/opentelemetry-go-contrib/pull/5279

This PR implements `Enabled` method on Core

### CODE REVIEW

File: bridges/otelzap/core.go [hunk 1]
```diff
@@ -107,10 +107,12 @@ func NewCore(opts ...Option) *Core {
 	}
 }
 
-// TODO
 // LevelEnabler decides whether a given logging level is enabled when logging a message.
 func (o *Core) Enabled(level zapcore.Level) bool {
-	return true
+	r := log.Record{}
+	r.SetSeverity(convertLevel(level))
+	// TODO How to pass context to Enabled method
```
Comment (pellared) [inline bridges/otelzap/core.go:114]: You cannot 😉 

```suggestion
```

File: bridges/otelzap/core_test.go [hunk 2]
```diff
@@ -31,6 +32,18 @@ func TestCore(t *testing.T) {
...
+	enabledFunc := func(c context.Context, r log.Record) bool {
+		return r.Severity() >= log.SeverityInfo
+	}
+
+	r := logtest.NewRecorder(logtest.WithEnabledFunc(enabledFunc))
+	zc := NewCore(WithLoggerProvider(r))
+
+	assert.False(t, zc.Enabled(zap.DebugLevel), "level conversion: permissive")
+	assert.True(t, zc.Enabled(zap.InfoLevel), "level conversion: restrictive")
+}
```
Comment (pellared) [inline bridges/otelzap/core_test.go:45]: Can you please test the functionality in a producation-like way (like in `TestCore`) by using https://pkg.go.dev/go.uber.org/zap#Logger.Check?

File: bridges/otelzap/core_test.go [hunk 3]
```diff
@@ -31,6 +33,29 @@ func TestCore(t *testing.T) {
...
+	}
+
+	r := logtest.NewRecorder(logtest.WithEnabledFunc(enabledFunc))
+	logger := zap.New(NewCore(WithLoggerProvider(r)))
+
+	if ce := logger.Check(zap.DebugLevel, testMessage); ce != nil {
+		ce.Write()
+	}
+
+	assert.Nil(t, r.Result()[1].Records)
```
Comment (pellared) [inline bridges/otelzap/core_test.go:48]:
```suggestion
	assert.Empty(t, r.Result()[1].Records)
```

File: bridges/otelzap/core_test.go [hunk 4]
```diff
@@ -31,6 +33,29 @@ func TestCore(t *testing.T) {
...
+
+	if ce := logger.Check(zap.InfoLevel, testMessage); ce != nil {
+		ce.Write()
+	}
+
+	got := r.Result()[1].Records[0]
```
Comment (pellared) [inline bridges/otelzap/core_test.go:54]: let's make sure that this test will not panic if the implementation does not work properly

```suggestion
	require.Len(t, r.Result()[1].Records, 1)
	got := r.Result()[1].Records[0]
```

File: bridges/otelzap/core_test.go [hunk 5]
```diff
@@ -31,6 +33,29 @@ func TestCore(t *testing.T) {
 	assert.Equal(t, log.SeverityInfo, got.Severity())
 }
 
+func TestCoreEnabled(t *testing.T) {
+	enabledFunc := func(c context.Context, r log.Record) bool {
+		return r.Severity() >= log.SeverityInfo
+	}
+
+	r := logtest.NewRecorder(logtest.WithEnabledFunc(enabledFunc))
```
[the remark this answers is not in the document]
Comment (khushijain21) [reply bridges/otelzap/core_test.go:41]: yes, but I was thinking from the user perspective. There is no WithEnabledFunc method available on the SDK implementation of `LoggerProvider` right?

[the remark this answers is not in the document]
Comment (pellared) [reply bridges/otelzap/core_test.go:41]: Right now the user would have to create a custom [`log.Processor`](https://pkg.go.dev/go.opentelemetry.io/otel/sdk/log#Processor). We have an issue to add a "minimum-level-processor":  https://github.com/open-telemetry/opentelemetry-go/issues/5078. However, I think it is better to postpone it before we have clarification how to do it in the specification: https://github.com/open-telemetry/opentelemetry-specification/issues/3917

File: bridges/otelzap/core_test.go [hunk 6]
```diff
@@ -31,6 +33,30 @@ func TestCore(t *testing.T) {
 	assert.Equal(t, log.SeverityInfo, got.Severity())
 }
 
+func TestCoreEnabled(t *testing.T) {
```
Comment (pellared) [inline bridges/otelzap/core_test.go:36]: What about `logger.Debug(testMessage)`. Will it be recorded or not? I think it deserves a test case.

**Review by pellared — APPROVED** [review]

**Review by XSAM — APPROVED** [review]

**Pull request merged**